### PR TITLE
docs: extend log-http flag docs and better explain syntax of flags supporting multiple data sources

### PR DESCRIPTION
**Pull request by mmatczuk**
Relevant changes in flags

```
    --log-http [api|proxy:]<none|short-url|url|headers|body|errors>,...  (default errors) (env FORWARDER_LOG_HTTP)
        HTTP request and response logging mode. 
        
        Modes: 
        - none: no logging
        - short-url: logs [scheme://]host[/path] instead of the full URL
        - url: logs the full URL including query parameters
        - headers: logs request line and headers
        - body: logs request line, headers, and body
        - errors: logs request line and headers if status code is greater than or equal to 500
        
        Modes for different modules can be specified separated by commas. 
        Example: "--log-http=api:errors,proxy:headers,url" specifies that the API module logs errors, the proxy module
        logs headers, and anything else logs full URL.
```

```
    -p, --pac <path or URL> (env FORWARDER_PAC)
        Proxy Auto-Configuration file to use for upstream proxy selection. 
        
        Syntax:
        - File: /path/to/file.pac
        - URL: http://example.com/proxy.pac
        - Embed: data:base64,<base64 encoded data>
        - Stdin: -
```

```
    --tls-cert-file <path or base64> (env FORWARDER_TLS_CERT_FILE)
        TLS certificate to use if the server protocol is https or h2. 
        
        Syntax:
        - File: /path/to/file.pac
        - Embed: data:base64,<base64 encoded data>
```

### CODE REVIEW

File: docs/content/config/forwarder_run.yaml [hunk 1]
```diff
@@ -104,20 +110,21 @@
...
+# precedence.
 #proxy: 
 
 # proxy-header <header>
 #
-# Add or remove HTTP headers on the CONNECT request to the upstream proxy. See
-# the documentation for the -H, --header flag for more details on the format.
+# Use --connect-header flag instead.
+# 
+DEPRECATED: --connect-header
```
Comment (Choraden) [inline docs/content/config/forwarder_run.yaml:127]: Deprecated notice should be also commented, shouldn't it?

**Comment by Choraden** [comment]
Apart from the deprecated notice LGTM

File: docs/content/config/forwarder_run.yaml (hunk 1 above)
Comment (mmatczuk) [reply docs/content/config/forwarder_run.yaml:137]: 🤔

**Comment by mmatczuk** [comment]
@Choraden PTAL

**Comment by Choraden** [comment]
Screenshot 2024-05-21 at 11 36 50

`Embed: data:base64,<base64 encoded data>` is not rendering well

**Review by Choraden — APPROVED** [review]

**Pull request merged**